File: Easy/1704_Determine_if_String_Halves_Are_Alike/Solution.java


public class Solution {
    private boolean isVowel(char c) {
        return "aeiouAEIOU".indexOf(c) != -1;
    }

    public boolean halvesAreAlike(String s) {
        int n = s.length();
        int count = 0;

        for (int i = 0; i < n / 2; i++) {
            if (isVowel(s.charAt(i))) {
                count++;
            }
        }
        for (int i = n / 2; i < n; i++) {
            if (isVowel(s.charAt(i))) {
                count--;
            }
        }

        return count == 0;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        System.out.println(solution.halvesAreAlike("book")); // true

        // test cases 2
        System.out.println(solution.halvesAreAlike("textbook")); // false
        
    }
}

File: Easy/1704_Determine_if_String_Halves_Are_Alike/Solution.py
from typing import List, Optional


class Solution:
    def isVowel(self, ch: str) -> bool:
        return ch in 'aeiouAEIOU'
    
    def halvesAreAlike(self, s: str) -> bool:
        n = len(s)
        count = 0

        for i in range(n // 2):
            if self.isVowel(s[i]):
                count += 1
        
        for i in range(n // 2, n):
            if self.isVowel(s[i]):
                count -= 1
        
        return count == 0
            


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    print(solution.halvesAreAlike("book")) # true

    # test cases 2
    print(solution.halvesAreAlike("textbook")) # false
    